app/email/utils.go

```diff
@@ -8,9 +8,14 @@ import (
 	"github.com/mattermost/mattermost-server/v6/utils"
 )
 
-func (es *Service) mailServiceConfig() *mail.SMTPConfig {
+func (es *Service) mailServiceConfig(replyToAddress string) *mail.SMTPConfig {
 	emailSettings := es.config().EmailSettings
 	hostname := utils.GetHostnameFromSiteURL(*es.config().ServiceSettings.SiteURL)
+
+	if replyToAddress == "" {
+		replyToAddress = *emailSettings.ReplyToAddress
+	}
```

Comment:
If the replyToAddress is part of the config, why can we default this to the address we want for cloud and there would be no need to do a special case override here https://github.com/mattermost/enterprise/blob/6b3accafc357e037105fafe14643396708eb90b9/cloud/config/cloud_defaults.json#L16

Reply:
@marianunez I thought so too, but that means all cloud emails reply to would be that value set in the config. But based on this conversation https://mattermost.atlassian.net/browse/MM-42522?focusedCommentId=125238, Jason mentioned that it shouldn't be that way

Reply:
I see, but if we are overriding it here at the highest level of the Mail Service, wouldn't this change affect all emails anyways?

Reply:
I see now that we instantiate this for each email. I thought this was a shared config for all emails. 👍